### PR TITLE
fix(sso): Update and simplify SSO docs

### PR DESCRIPTION
**Pull request by BYK**
Fixes an [issue reported on the forum](https://forum.sentry.io/t/upgrade-9-0-with-github-sso-9-1-internal-server-error/8613?u=byk) and getsentry/onpremise#373.

### CODE REVIEW

File: src/collections/_documentation/server/sso.md [hunk 1]
```diff
@@ -8,54 +8,25 @@ SSO in Sentry is handled in one of two ways:
...
-# turn SSO on our off
-SENTRY_FEATURES['organizations:sso'] = False
-```
-
-Additionally you may enable advanced SSO features:
-
-```python
-from sentry.conf.server import *
-
-SENTRY_FEATURES['organizations:sso-saml2'] = True
```
Comment (scefali) [inline src/collections/_documentation/server/sso.md]: @BYK curious why we are removing the example of Saml2 and rippling?

Comment (BYK) [reply src/collections/_documentation/server/sso.md]: This is not an example, it is just showing a feature flag that is obsolete as it is enabled by default for a while now.

Comment (BYK) [reply src/collections/_documentation/server/sso.md]: @EvanPurkhiser it is obsolete for on-premise as it is enabled by default there.

File: src/collections/_documentation/server/sso.md [hunk 2]
```diff
@@ -8,54 +8,25 @@ SSO in Sentry is handled in one of two ways:
 -   Via a middleware which handles an upstream proxy dictating the authenticated user
 -   Via a third-party service which implements an authentication pipeline
 
-This documentation describes the latter, which would cover things like Google Apps, GitHub, LDAP, and other similar services.
+This documentation describes the latter, which would cover things like Google Apps and GitHub.
```
Comment (MimiDumpling) [inline src/collections/_documentation/server/sso.md]:
```suggestion
This documentation describes the latter, which will cover topics like Google Apps and GitHub.
```

File: src/collections/_documentation/server/sso.md [hunk 3]
```diff
@@ -8,54 +8,25 @@ SSO in Sentry is handled in one of two ways:
...
-
-The following providers are published and maintained by the Sentry team:
-
--   [Google Apps](https://github.com/getsentry/sentry-auth-google)
--   [GitHub](https://github.com/getsentry/sentry-auth-github)
-
 ## Custom Providers
 
-At this time the API is considered unstable and subject to change. Things likely won’t change a lot, but there’s a few areas that need cleaned up.
+At this time the API is considered unstable and subject to change. Things likely won’t change a lot, but there’s a few areas that need some clean up.
```
Comment (MimiDumpling) [inline src/collections/_documentation/server/sso.md]:
```suggestion
At this time, the API is considered unstable and subject to change. There won't be many changes, but a few areas need some cleanup.
```

**Pull request merged**